calico/common.py

```diff
@@ -547,3 +552,41 @@ def validate_tags(profile_id, tags):
 
     if issues:
         raise ValidationFailed(" ".join(issues))
+
+def validate_ipam_pool(pool_id, pool, ip_version):
+    # Remove any keys that we're not expecting.  Stops unvalidated data from
+    # slipping through.  We ignore other keys since this structure is used
+    # by calicoctl for its own purposes too.
+    keys_to_remove = set()
+    for key in pool:
+        if key not in ("cidr", "ipip", "masquerade"):
+            keys_to_remove.add(key)
+    for key in keys_to_remove:
+        pool.pop(key)
+
+    issues = []
+    if "cidr" not in pool:
+        issues.append("'cidr' field is missing")
+    else:
+        cidr = pool["cidr"]
+        if cidr is None or not validate_cidr(cidr, ip_version):
+            issues.append("Invalid CIDR: %r" % cidr)
+        else:
+            pool["cidr"] = canonicalise_cidr(cidr, ip_version)
+
+    if "ipip" in pool:
```

Comment:
We think we'll use this soon (to decide whether to turn on the IPIP function for a particular pool or not) but it's not yet used.